### PR TITLE
Add NATS Autopilot Pattern

### PR DESCRIPTION
**Pull request by geek**

### CODE REVIEW

**Comment by sberryman** [comment]
YES!!!! What are the plans for seneca and NATS?

**Comment by geek** [comment]
@sberryman no direct plans for seneca-nats, but happy to help if you have plans.

**Comment by sberryman** [comment]
@geek, honestly today is the first I've heard of NATS and just got done watching Derek Collison's talk on it. My biggest problem right now with seneca is the req/res assumption; this is also not the right place to be discussing seneca. Either way, it would be interesting to hear how you guys are using NATS. I'm currently using seneca with rabbitmq as the messaging layer. I tried to go with [SWIM](https://www.npmjs.com/package/seneca-mesh) but it was easier for me to hack together observers on top of rabbitmq topic exchanges.

**Comment by geek** [comment]
@sberryman I like the idea of using NATS in the same way you would use rabbitmq with seneca today. Another actor model solution that uses NATS is [hemera](https://github.com/hemerajs/hemera). 

There is a publish only mechanism in seneca, if you don't expect a response... `seneca.act({ my: 'message' });` <-- no need for the reply callback.

**Comment by sberryman** [comment]
Any idea when you are going to merge this? I ended up getting a few of my old colleagues at Golf Channel on the NATS bandwagon after seeing you post this. I'm also seriously considering a quick attempt at a seneca transport. RMQ is overkill for what I need right now.

**Comment by geek** [comment]
@sberryman very cool. I am hoping to have this merged and published tomorrow. How does this PR look to you, anything that you would like to see?

File: bin/manage.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+consulCommand() {
+    consul-cli --quiet --consul="${CONSUL}:8500" $*
```
Comment (tgross) [inline bin/manage.sh]: What's this getting us that we don't already have from the `consul` binary in the container?

File: makefile [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+# ------------------------------------------------
+# Test running
+
+## Pull the container images from the Docker Hub
+pull:
+	docker pull $(image):$(tag)
+
+$(DOCKER_CERT_PATH)/key.pub:
+	ssh-keygen -y -f $(DOCKER_CERT_PATH)/key.pem > $(DOCKER_CERT_PATH)/key.pub
```
Comment (tgross) [inline makefile:53]: You should try setting up the test job on Jenkins... I'm fairly certain we don't need this here.

File: bin/manage.sh (hunk 1 above)
Comment (geek) [reply bin/manage.sh]: @tgross good catch, I removed in favor of `consul members`

File: makefile (hunk 2 above)
Comment (geek) [reply makefile:53]: @tgross I will do that, I was hoping to get an initial version out to be able to have the tests fully working

**Pull request merged**